controllers/openshift_controller.go

```diff
@@ -1727,14 +1742,119 @@ func (r *KataConfigOpenShiftReconciler) enablePeerPods() error {
 	if k8serrors.IsAlreadyExists(err) {
 		return nil
 	} else if err != nil {
 		return err
 	}
 
-	err = r.createRuntimeClass("kata-remote-cc", "0.25", "350Mi")
+	//Create MachineConfig for kata-remote hyp CRIO config
+	err = r.createMcFromFile(peerpodsCrioMachineConfigYaml)
+	if err != nil {
+		r.Log.Error(err, "Unable to create MachineConfig ", "peerpodsCrioMachineConfig", peerpodsCrioMachineConfig)
+		return err
+	}
 
-	return err
+	//Create MachineConfig for kata-remote hyp config toml
+	err = r.createMcFromFile(peerpodsKataRemoteMachineConfigYaml)
+	if err != nil {
+		r.Log.Error(err, "Unable to create MachineConfig ", "peerpodsKataRemoteMachineConfig", peerpodsKataRemoteMachineConfig)
+		return err
+	}
+
+	// Create runtimeClass config for peer-pods
+	err = r.createRuntimeClass(peerpodsRuntimeClassName, peerpodsRuntimeClassCpuOverhead, peerpodsRuntimeClassMemOverhead)
+	if err != nil {
+		r.Log.Error(err, "Unable to create RuntimeClass ", "peerpodsRuntimeClassName", peerpodsRuntimeClassName)
+		return err
+	}
+	return nil
+}
+
+func (r *KataConfigOpenShiftReconciler) disablePeerPods() error {
+	peerpodconf := v1alpha1.PeerPodConfig{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      peerpodConfigCrdName,
+			Namespace: "openshift-sandboxed-containers-operator",
+		},
+	}
+	err := r.Client.Delete(context.TODO(), &peerpodconf)
+	if err != nil {
+		// error during removing peerpodconfig. Just log the error and move on.
+		r.Log.Error(err, "Error found deleting PeerPodConfig. If the PeerPodConfig object exists after uninstallation it can be safely deleted manually")
+	}
+
+	mc := mcfgv1.MachineConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "machineconfiguration.openshift.io/v1",
+			Kind:       "MachineConfig",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: peerpodsKataRemoteMachineConfig,
+		},
+	}
+
+	err = r.Client.Delete(context.TODO(), &mc)
+	if err != nil {
+		// error during removing mc. Just log the error and move on.
+		r.Log.Error(err, "Error found deleting mc. If the MachineConfig object exists after uninstallation it can be safely deleted manually",
+			"mc", mc.Name)
+	}
+
+	mc = mcfgv1.MachineConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "machineconfiguration.openshift.io/v1",
+			Kind:       "MachineConfig",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: peerpodsCrioMachineConfig,
+		},
+	}
+
+	err = r.Client.Delete(context.TODO(), &mc)
+	if err != nil {
+		// error during removing mc. Just log the error and move on.
+		r.Log.Error(err, "Error found deleting mc. If the MachineConfig object exists after uninstallation it can be safely deleted manually",
+			"mc", mc.Name)
+	}
+
+	return nil
+}
+
+func (r *KataConfigOpenShiftReconciler) createMcFromFile(mcFileName string) error {
+	yamlData, err := readMachineConfigYAML(mcFileName)
+	if err != nil {
+		r.Log.Error(err, "Unable to read MachineConfigYaml ", "mcFileName", mcFileName)
```

Comment:
nit: generally speaking, why not returning formatted error instead

Reply:
The `err` is already formatted. Just logging it here for convenience.